notebooks/MAST/HSC/CASJOBS/SWEEPS_HSCV3P1/sweeps_hscv3p1.ipynb

```diff
@@ -0,0 +1,2063 @@
+    "pip install mastcasjobs\n",
+    "```\n",
+    "\n",
+    "### Set up your CasJobs account information\n",
+    "\n",
+    "You must have a [MAST Casjobs][3] account. Note that MAST Casjobs accounts are independent of SDSS Casjobs accounts.\n",
+    "\n",
+    "For easy startup, you can optionally set the environment variables `CASJOBS_USERID` and/or `CASJOBS_PW` with your Casjobs account information.  The Casjobs user ID and password are what you enter when logging into Casjobs.\n",
+    "\n",
+    "This script prompts for your Casjobs user ID and password during initialization if the environment variables are not defined.\n",
+    "\n",
+    "\n",
+    "  [1]: https://www.anaconda.com/products/distribution\n",
+    "  [2]: https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/channels.html\n",
+    "  [3]: https://mastweb.stsci.edu/hcasjobs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import astropy\n",
+    "import time\n",
+    "import sys\n",
+    "import os\n",
+    "import requests\n",
+    "import numpy as np\n",
+    "from matplotlib.colors import LogNorm\n",
+    "import matplotlib.pyplot as plt\n",
+    "from pathlib import Path\n",
+    "\n",
+    "# check that version of mastcasjobs is new enough\n",
+    "# we are using some features not in version 0.0.1\n",
+    "from pkg_resources import get_distribution\n",
+    "from packaging.version import Version as V\n",
+    "\n",
+    "assert V(get_distribution(\"mastcasjobs\").version) >= V('0.0.2'), \"\"\"\n",
+    "A newer version of mastcasjobs is required.\n",
+    "Update mastcasjobs to current version using this command:\n",
+    "pip install --upgrade git+git://github.com/rlwastro/mastcasjobs@master\n",
+    "\"\"\"\n",
+    "\n",
+    "import mastcasjobs\n",
+    "\n",
+    "## For handling ordinary astropy Tables\n",
+    "from astropy.table import Table\n",
+    "from astropy.io import fits, ascii\n",
+    "\n",
+    "from fastkde import fastKDE\n",
+    "from scipy.interpolate import RectBivariateSpline\n",
+    "from astropy.modeling import models, fitting\n",
+    "\n",
+    "# There are a number of relatively unimportant warnings that \n",
+    "# show up, so for now, suppress them:\n",
+    "import warnings\n",
+    "warnings.filterwarnings(\"ignore\")\n",
+    "\n",
+    "# set width for pprint\n",
+    "astropy.conf.max_width = 150"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set universal matplotlib parameters\n",
+    "plt.rcParams.update({'font.size': 16})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# directory where generated plots are saved\n",
+    "resPath = Path(\"./Plots/sweeps_hscv3p1\")\n",
+    "resPath.mkdir(parents=True, exist_ok=True)\n",
+    "HSCContext = \"HSCv3\""
```

Comment:
There was existing code to save certain files. Since I hade all notebooks in the same directory, as other people might, it made sense to have the those images saved to a location for that specific notebook. If nothing else, it shows some of the usefulness of `pathlib`, but it can certainly be removed.

Reply:
done